src/main/kotlin/org/elm/lang/core/resolve/scope/GlobalScope.kt

```diff
@@ -68,57 +77,82 @@ class GlobalScope private constructor(val clientFile: ElmFile) {
-        rest.addAll(helper("Basics"))
-        rest.addAll(helper("List").filter { it.name == "::" })
+        private fun produceVisibleValues(clientFile: ClientLocation): List<ElmNamedElement> {
+            fun helper(moduleName: String) =
+                    ElmModulesIndex.get(moduleName, clientFile)
+                            ?.let { ModuleScope.getDeclaredValues(it.elmFile) }
+
+            val rest = mutableListOf<ElmNamedElement>()
+            helper("Basics")?.list?.let { rest.addAll(it) }
+            helper("List")?.get("::")?.let { rest.add(it) }
+
+            // TODO [drop 0.18] remove this line (the `!` operator was removed in 0.19)
+            helper("Platform.Cmd")?.get("!")?.let { rest.add(it) }
+
+            return rest
+        }
+
+
+        // this function should also return List, but, unlike all the other global types, Elm doesn't
+        // define it anywhere, so there's no element to return
+        private fun produceVisibleTypes(location: ClientLocation): List<ElmNamedElement> {
+            fun helper(moduleName: String) =
+                    ElmModulesIndex.get(moduleName, location)
+                            ?.let { ModuleScope.getDeclaredTypes(it.elmFile) }
+
+            val rest = mutableListOf<ElmNamedElement>()
+            helper("Basics")?.list?.let { rest.addAll(it) }
+            helper("Char")?.get("Char")?.let { rest.add(it) }
+            helper("Maybe")?.get("Maybe")?.let { rest.add(it) }
+            helper("Result")?.get("Result")?.let { rest.add(it) }
+            helper("String")?.get("String")?.let { rest.add(it) }
+            helper("Platform")?.get("Program")?.let { rest.add(it) }
+            helper("Platform.Cmd")?.get("Cmd")?.let { rest.add(it) }
+            helper("Platform.Sub")?.get("Sub")?.let { rest.add(it) }
+            return rest
+        }
+
+        private fun produceVisibleConstructors(location: ClientLocation): List<ElmNamedElement> {
+            fun helper(moduleName: String) =
+                    ElmModulesIndex.get(moduleName, location)
+                            ?.let { ModuleScope.getDeclaredConstructors(it.elmFile) }
 
-        // TODO [drop 0.18] remove this line (the `!` operator was removed in 0.19)
-        rest.addAll(helper("Platform.Cmd").filter { it.name == "!" })
+            val rest = mutableListOf<ElmNamedElement>()
+            helper("Basics")?.list?.let { rest.addAll(it) }
+            helper("Maybe")?.let { ctors ->
+                ctors["Just"]?.let { rest.add(it) }
+                ctors["Nothing"]?.let { rest.add(it) }
+            }
+            helper("Result")?.let { ctors ->
+                ctors["Ok"]?.let { rest.add(it) }
+                ctors["Err"]?.let { rest.add(it) }
+            }
+            return rest
+        }
+    }
 
-        return rest
+    fun getVisibleValues(): List<ElmNamedElement> {
+        val elmProject = clientFile.elmProject ?: return produceVisibleValues(clientFile)
```

Comment:
If the `elmProject` property is null, just return early with an empty list. We cannot provide good results without it.

Ditto for `getVisibleTypes()` and `getVisibleConstructors()`